paddle/framework/block_desc.h

```diff
@@ -57,10 +57,16 @@ class BlockDescBind {
 
   OpDescBind *AppendOp();
 
+  void AppendAllocatedOp(std::unique_ptr<OpDescBind> &op_desc);
+
   OpDescBind *PrependOp();
 
   std::vector<OpDescBind *> AllOps() const;
 
+  int OpSize() const { return ops_.size(); }
+
+  OpDescBind *Op(int idx) const { return ops_.at(idx).get(); }
```

Comment:
use size_t as idx, also maybe do not mark this method as `const` since it will change the program.

Reply:
Done